src/CSnakes.SourceGeneration/PythonStaticGenerator.cs

```diff
@@ -119,8 +141,13 @@ public void Dispose()
             }
             public interface I{{pascalFileName}} : IDisposable
             {
+                /// <summary>
+                /// Reload the module, useful for development if the Python code has changed.
+                /// </summary>
+                public void ReloadModule();
```

Comment:
This ought to be inherited through an interface in `CSnakes.Runtime` along the lines of:

```c#
public interface IReloadable : IDisposable
{
    /// <summary>
    /// Reload the module, useful for development if the Python code has changed.
    /// </summary>
    public void ReloadModule();
}
```

rather than being generated. The name is up for debate. While the chances are super low, this will also help to avoid the potential for a compilation error if the source Python module also contains a top-level function named `reload_module`; the implementing can [explicitly implement](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/interfaces/explicit-interface-implementation) the interface:

```c#
void IModuleProxy.ReloadModule() { /* ... */ }
```

In the absolute worst case, the user will get the [CS0108](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0108) warning instead of a compilation error. In fact, the source generator could add the `new` keyword if such a conflict exists to avoid a CS0108 warning altogether.

Reply:
It's a static class, so I'm not sure how you could implement this one simply?

Reply:
See PR #269 as a demonstration of this.